actions/poweroff/v1/poweroff.go

```diff
@@ -0,0 +1,43 @@
+package main
+
+import (
+    "fmt"
+    "log"
+    "os/exec"
+    "runtime"
+)
+
+// Dont have global variable !
+
+func main() {
+    // Checking the machine system
+    if (runtime.GOOS == "linux") {
+        LinuxShutdown()
+    } else if (runtime.GOOS == "windows") {
+        WindowsShutdown()
+    } else {
+        fmt.Println("Sorry, this action cannot run on this server!")
+    }
+}
+
+// WindowsShutdown is for windows
+func WindowsShutdown() {
+    output, err := exec.Command("shutdown").Output()
+    // Handel error
+    if (err != nil) {
+        log.Fatalln("Some thing went wrong:", err)
+    }
+    // Shut down system
+    result := string(output)
+    fmt.Println(result)
+}
+
+// LinuxShutdown is for linux
+func LinuxShutdown() {
+    output, err := exec.Command("shutdown -h Now").Output()
```

Comment:
Out of curiosity, how did you test this action? The scratch image does not have a shutdown command, and I don't see it as part of your Dockerfile a place where you statically build it